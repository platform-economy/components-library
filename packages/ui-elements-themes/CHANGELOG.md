# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
