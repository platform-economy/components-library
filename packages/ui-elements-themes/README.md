# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
