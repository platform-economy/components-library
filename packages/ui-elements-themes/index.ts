export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
