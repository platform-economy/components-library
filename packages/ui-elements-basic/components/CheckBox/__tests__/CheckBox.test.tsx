import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

import { withTestTheme } from '@relayr/ui-elements-themes';
import * as TestRenderer from 'react-test-renderer';

enzyme.configure({ adapter: new Adapter() });

import { CheckBox } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('CheckBox', () => {
      it('Should handle click event correctly', () => {
        const onClick = jest.fn();
        const onValueChange = jest.fn();
        const onChange = jest.fn();
        const tree = enzyme.mount(withTestTheme(
          <CheckBox
            value={true}
            onClick={onClick}
            onValueChange={onValueChange}
            onChange={onChange}
          />));
        tree.find('span').simulate('click');
        tree.find('input').simulate('click');

        expect(onClick).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledTimes(0);
        expect(onValueChange).toHaveBeenCalledTimes(0);

        tree.find('input').simulate('change');
        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onValueChange).toHaveBeenCalledTimes(1);
        expect(onValueChange).toHaveBeenCalledWith(false);
      });

      it('Render Check', () => {
        const tree = enzyme.mount(withTestTheme(
          <CheckBox
            value="true"
          />));
        tree.find('input').simulate('change');
      });

      it('Render unCheck', () => {
        const tree = enzyme.mount(withTestTheme(
          <CheckBox
            value="false"
          />));
        tree.find('span').simulate('click');
      });

      it('Render disabled', () => {
        const tree = enzyme.mount(withTestTheme(
          <CheckBox
            value="true"
            disabled={true}
          />));
        tree.find('span').simulate('click');
      });

      it(' Render empty', () => {
        const tree = TestRenderer.create(withTestTheme(<CheckBox/>));
        expect(tree).toMatchSnapshot();
      });

      it('Render indeterminate and disabled', () => {
        const tree = TestRenderer.create(withTestTheme(
          <CheckBox
            value="indeterminate"
            disabled={true}
          />)).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Render check', () => {
        const tree = TestRenderer.create(withTestTheme(
          <CheckBox
            value="check"
          />)).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Render uncheck', () => {
        const tree = TestRenderer.create(withTestTheme(
          <CheckBox
            value="uncheck"
          />)).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
