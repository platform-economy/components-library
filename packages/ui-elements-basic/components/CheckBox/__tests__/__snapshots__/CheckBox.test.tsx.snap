// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components CheckBox  Render empty 1`] = `
.c0 {
  cursor: pointer;
}

.c0.disabled * {
  color: #inactive;
  cursor: not-allowed;
}

.c0 > input {
  display: none;
}

.c0 > svg {
  margin: 0px 5px;
  width: 1em;
  height: 1em;
}

.c0 > svg > path.checkbox-button-bg {
  stroke-width: 0;
  stroke: #active;
  fill: #active;
  display: none;
}

.c0.checked > svg > path.checkbox-button-bg,
.c0.indeterminate > svg > path.checkbox-button-bg {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-bg,
.c0:focus > svg > path.checkbox-button-bg {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-bg {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-bg {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-border {
  stroke-width: 0;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.unchecked > svg > path.checkbox-button-border {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-border,
.c0:focus > svg > path.checkbox-button-border {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-border {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-border {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-selection {
  stroke-width: 3;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.checked > svg > path.checkbox-button-selection {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-selection {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-schrodinger {
  stroke-width: 3;
  stroke: #text;
  fill: #text;
  display: none;
}

.c0.indeterminate > svg > path.checkbox-button-schrodinger {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-schrodinger {
  stroke: #inactive;
  fill: #inactive;
}

<span
  className="indeterminate c0"
  onClick={[Function]}
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    viewBox="-144 -144 624 624"
  >
    <path
      className="checkbox-button-bg"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48z"
    />
    <path
      className="checkbox-button-border"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352z"
    />
    <path
      className="checkbox-button-selection"
      d="m 310.864 241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
    />
    <path
      className="checkbox-button-schrodinger "
      d="M 92 296 c -6.6 0 -12 -5.4 -12 -12 v -56 c 0 -6.6 5.4 -12 12 -12 h 264 c 6.6 0 12 5.4 12 12 v 56 c 0 6.6 -5.4 12 -12 12 H 92 Z"
    />
  </svg>
</span>
`;

exports[`ui-elements-basic Components CheckBox Render check 1`] = `
.c0 {
  cursor: pointer;
}

.c0.disabled * {
  color: #inactive;
  cursor: not-allowed;
}

.c0 > input {
  display: none;
}

.c0 > svg {
  margin: 0px 5px;
  width: 1em;
  height: 1em;
}

.c0 > svg > path.checkbox-button-bg {
  stroke-width: 0;
  stroke: #active;
  fill: #active;
  display: none;
}

.c0.checked > svg > path.checkbox-button-bg,
.c0.indeterminate > svg > path.checkbox-button-bg {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-bg,
.c0:focus > svg > path.checkbox-button-bg {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-bg {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-bg {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-border {
  stroke-width: 0;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.unchecked > svg > path.checkbox-button-border {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-border,
.c0:focus > svg > path.checkbox-button-border {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-border {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-border {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-selection {
  stroke-width: 3;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.checked > svg > path.checkbox-button-selection {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-selection {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-schrodinger {
  stroke-width: 3;
  stroke: #text;
  fill: #text;
  display: none;
}

.c0.indeterminate > svg > path.checkbox-button-schrodinger {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-schrodinger {
  stroke: #inactive;
  fill: #inactive;
}

<span
  className="checked c0"
  onClick={[Function]}
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    viewBox="-144 -144 624 624"
  >
    <path
      className="checkbox-button-bg"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48z"
    />
    <path
      className="checkbox-button-border"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352z"
    />
    <path
      className="checkbox-button-selection"
      d="m 310.864 241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
    />
    <path
      className="checkbox-button-schrodinger "
      d="M 92 296 c -6.6 0 -12 -5.4 -12 -12 v -56 c 0 -6.6 5.4 -12 12 -12 h 264 c 6.6 0 12 5.4 12 12 v 56 c 0 6.6 -5.4 12 -12 12 H 92 Z"
    />
  </svg>
</span>
`;

exports[`ui-elements-basic Components CheckBox Render indeterminate and disabled 1`] = `
.c0 {
  cursor: pointer;
}

.c0.disabled * {
  color: #inactive;
  cursor: not-allowed;
}

.c0 > input {
  display: none;
}

.c0 > svg {
  margin: 0px 5px;
  width: 1em;
  height: 1em;
}

.c0 > svg > path.checkbox-button-bg {
  stroke-width: 0;
  stroke: #active;
  fill: #active;
  display: none;
}

.c0.checked > svg > path.checkbox-button-bg,
.c0.indeterminate > svg > path.checkbox-button-bg {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-bg,
.c0:focus > svg > path.checkbox-button-bg {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-bg {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-bg {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-border {
  stroke-width: 0;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.unchecked > svg > path.checkbox-button-border {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-border,
.c0:focus > svg > path.checkbox-button-border {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-border {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-border {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-selection {
  stroke-width: 3;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.checked > svg > path.checkbox-button-selection {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-selection {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-schrodinger {
  stroke-width: 3;
  stroke: #text;
  fill: #text;
  display: none;
}

.c0.indeterminate > svg > path.checkbox-button-schrodinger {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-schrodinger {
  stroke: #inactive;
  fill: #inactive;
}

<span
  className="disabled indeterminate c0"
  disabled={true}
  onClick={[Function]}
>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    viewBox="-144 -144 624 624"
  >
    <path
      className="checkbox-button-bg"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48z"
    />
    <path
      className="checkbox-button-border"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352z"
    />
    <path
      className="checkbox-button-selection"
      d="m 310.864 241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
    />
    <path
      className="checkbox-button-schrodinger "
      d="M 92 296 c -6.6 0 -12 -5.4 -12 -12 v -56 c 0 -6.6 5.4 -12 12 -12 h 264 c 6.6 0 12 5.4 12 12 v 56 c 0 6.6 -5.4 12 -12 12 H 92 Z"
    />
  </svg>
</span>
`;

exports[`ui-elements-basic Components CheckBox Render uncheck 1`] = `
.c0 {
  cursor: pointer;
}

.c0.disabled * {
  color: #inactive;
  cursor: not-allowed;
}

.c0 > input {
  display: none;
}

.c0 > svg {
  margin: 0px 5px;
  width: 1em;
  height: 1em;
}

.c0 > svg > path.checkbox-button-bg {
  stroke-width: 0;
  stroke: #active;
  fill: #active;
  display: none;
}

.c0.checked > svg > path.checkbox-button-bg,
.c0.indeterminate > svg > path.checkbox-button-bg {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-bg,
.c0:focus > svg > path.checkbox-button-bg {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-bg {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-bg {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-border {
  stroke-width: 0;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.unchecked > svg > path.checkbox-button-border {
  display: inline;
}

.c0:hover > svg > path.checkbox-button-border,
.c0:focus > svg > path.checkbox-button-border {
  stroke: #active_hovered;
  fill: #active_hovered;
}

.c0:active > svg > path.checkbox-button-border {
  stroke: #active_lowered;
  fill: #active_lowered;
}

.c0.c0.disabled > svg > path.checkbox-button-border {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-selection {
  stroke-width: 3;
  stroke: #text_highlight;
  fill: #text_highlight;
  display: none;
}

.c0.checked > svg > path.checkbox-button-selection {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-selection {
  stroke: #inactive;
  fill: #inactive;
}

.c0 > svg > path.checkbox-button-schrodinger {
  stroke-width: 3;
  stroke: #text;
  fill: #text;
  display: none;
}

.c0.indeterminate > svg > path.checkbox-button-schrodinger {
  display: inline;
}

.c0.c0.disabled > svg > path.checkbox-button-schrodinger {
  stroke: #inactive;
  fill: #inactive;
}

<span
  className="unchecked c0"
  onClick={[Function]}
>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    viewBox="-144 -144 624 624"
  >
    <path
      className="checkbox-button-bg"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48z"
    />
    <path
      className="checkbox-button-border"
      d="M400 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zm0 400H48V80h352v352z"
    />
    <path
      className="checkbox-button-selection"
      d="m 310.864 241.724L191.547 361.48c-4.705 4.667-12.303 4.637-16.97-.068l-90.781-91.516c-4.667-4.705-4.637-12.303.069-16.971l22.719-22.536c4.705-4.667 12.303-4.637 16.97.069l59.792 60.277 141.352-140.216c4.705-4.667 12.303-4.637 16.97.068l22.536 22.718c4.667 4.706 4.637 12.304-.068 16.971z"
    />
    <path
      className="checkbox-button-schrodinger "
      d="M 92 296 c -6.6 0 -12 -5.4 -12 -12 v -56 c 0 -6.6 5.4 -12 12 -12 h 264 c 6.6 0 12 5.4 12 12 v 56 c 0 6.6 -5.4 12 -12 12 H 92 Z"
    />
  </svg>
</span>
`;
